### PR TITLE
Allow custom EHLO after RSET again

### PR DESCRIPTION
**Pull request by sapmli**
Resolves [EHLO after RSET #269](https://github.com/emersion/go-smtp/issues/269)

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -229,14 +229,25 @@ func (c *Conn) handleGreet(enhanced bool, arg string) {
...
-	if err != nil {
-		c.helo = ""
-		c.writeError(451, EnhancedCode{4, 0, 0}, err)
-		return
-	}
+	// RFC 5321: "An EHLO command MAY be issued by a client later in the session"
+	if c.session != nil {
+		// RFC 5321: "If it is issued after the session begins
+		// and the EHLO command is acceptable to the SMTP server, ..."
+		if !c.fromReceived {
```
Comment (emersion) [inline conn.go:236]: Hm, I don't quite get why this condition is here. Why do we need to skip `reset()` when the client has sent `MAIL FROM`?

Comment (sapmli) [reply conn.go:236]: It's what I found in the session that may be usable for the condition "if it is issued after the session begins"

Comment (emersion) [reply conn.go:236]: ["Session begins"](https://datatracker.ietf.org/doc/html/rfc5321#section-3.1) just means that a TCP connection is opened no?

Comment (sapmli) [reply conn.go:236]: Intersting question. If so, then do you think that an EHLO after RSET should not be passed to the session as Reset()?

Comment (emersion) [reply conn.go:236]: My point is that we should probably always call `c.reset()`, even in the case where `MAIL FROM` was sent by the client (that's what `fromReceived` means).

Comment (sapmli) [reply conn.go:236]: Okay, I can do that. I'll update the PR.

Comment (sapmli) [reply conn.go]: But feel free to adapt it to your likings and code style etc.

**Review by emersion — APPROVED** [review]
Thanks!

I've re-organized the changes into two commits.

**Pull request merged**